core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2137,7 +2138,7 @@ public static String getFilePathExternalFilePath(String path, Configuration conf
     if (fistFilePath == null) {
       // Check if we can infer the schema from the hive metastore.
       LOGGER.error("CarbonData file is not present in the table location");
-      throw new IOException("CarbonData file is not present in the table location");
+      throw new FileNotFoundException("CarbonData file is not present in the table location");
```

Comment:
@Karan980 , the inferSchema is called from many places, basically can you check and  confirm from the code that, when you throw the FileNotFoundException, the exception is properly handled from all the callers or callers of caller and confirm no where the exception will be hidden due to this change.

Reply:
@akashrn5 yes inferSchema is called from several places. i have checked it direct callers and callers of callers. This change is not effecting other calls of inferSchema.